web/src/drafts_overlay_ui.ts

```diff
@@ -21,6 +22,57 @@ import {realm} from "./state_data.ts";
 import * as user_card_popover from "./user_card_popover.ts";
 import * as user_group_popover from "./user_group_popover.ts";
 
+let draft_undo_delete_list: LocalStorageDraft[] = [];
+
+function clear_undo_list(): void {
+    draft_undo_delete_list = [];
+    $("#draft_overlay_banner_container").empty();
+}
+
+function undo_draft_deletion(): void {
+    if (draft_undo_delete_list.length === 0) {
+        return;
+    }
+
+    for (const draft of draft_undo_delete_list) {
+        drafts.draft_model.addDraft(draft);
+    }
+
+    clear_undo_list();
+    rerender_drafts();
+}
+
+function show_delete_banner(deleted_count: number): void {
+    const $banner_container = $("#draft_overlay_banner_container");
+    $banner_container.empty();
+    const banner_html = render_banner({
+        intent: "success",
+        label: $t(
+            {
+                defaultMessage:
+                    "{N, plural, one {# draft was deleted.} other {# drafts were deleted.}}",
+            },
+            {N: deleted_count},
+        ),
+        buttons: [
+            {
+                attention: "quiet",
+                intent: "success",
+                label: $t({defaultMessage: "Undo"}),
+            },
+        ],
+        close_button: true,
+    });
+
+    $banner_container.html(banner_html);
+    $banner_container
+        .off("click", ".action-button-quiet-success")
+        .on("click", ".action-button-quiet-success", undo_draft_deletion);
+    $banner_container
+        .off("click", ".banner-close-button")
+        .on("click", ".banner-close-button", clear_undo_list);
```

Comment:
I don't think you need set up this click handler every time. Can you put it in this file's `initialize` function?